### ISSUE TITLE
Integration with the Angular CLI

### ISSUE THREAD

**Issue by MikeRyanDev** [synapse-wireless-labs/component-lab#6]
By default, Component Lab should try to resolve the webpack configuration from angular-cli if it detects it is installed.

**Comment by isaacplmann** [comment]
PR #12 gets part of the way there.  It's not really a sure fire method, since I'm accessing internals of the cli.

**Comment by intellix** [comment]
Was just having a go at implementing this myself with CLI 1.0.0.
I created a separate app within `.angular-cli.json` pointing to a new entrypoint: **lab.ts** and added the `createLab({...})` declared in README.md.

Got an error that: `Tried to find bootstrap code, but could not. Specify either statically analyzable bootstrap code or pass in an entryModule to the plugins options.`

Because CLI is searching the entrypoint for a `platformBrowserDynamic.bootstrapModule` function to be declared there, which is defined here: https://github.com/synapse-wireless-labs/component-lab/blob/master/src/frontend/bootstrap.ts

Either this module needs to expose ComponentLabModule so we can bootstrap this ourselves or CLI needs to support detecting dynamically loaded bootstrap functions.

**Comment by intellix** [comment]
The problem is that the bootstrap procedure isn't statically analysable. This works with CLI 1.0.0:

Add an extra app to **/.angular-cli.json**
```
    {
      "name": "lab",
      "root": "src",
      "outDir": "dist",
      "assets": [
        "assets",
        "favicon.ico"
      ],
      "index": "lab.html",
      "main": "lab.ts",
      "polyfills": "polyfills.ts",
      "test": "test.ts",
      "tsconfig": "tsconfig.app.json",
      "testTsconfig": "tsconfig.spec.json",
      "prefix": "app",
      "styles": [
        "styles.css"
      ],
      "scripts": [],
      "environmentSource": "environments/environment.ts",
      "environments": {
        "dev": "environments/environment.ts",
        "prod": "environments/environment.prod.ts"
      }
    }
```

**/src/lab.ts**
```ts
import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
import { getModuleForExperiments } from 'component-lab/src/frontend/module-builder';
import { provideExperiments } from 'component-lab/src/frontend/services/experiment-registry';
import { provideResolvedLab } from 'component-lab/src/frontend/services/experiment-factory';
import { ComponentLabModule } from 'component-lab/src/frontend/component-lab.module';

import { AppModule } from './app/app.module';

declare var require: any;

const loadExperiments = () => {
  const context = require.context('./', true, /\.exp\.ts/);
  return context.keys().map(context).map(mod => mod.default);
};

const experiments = loadExperiments();
const resolved = getModuleForExperiments(AppModule, experiments);

platformBrowserDynamic([
  provideExperiments(experiments),
  provideResolvedLab(resolved)
]).bootstrapModule(ComponentLabModule);
```

**/src/lab.html**
```html
<!doctype html>
<html>
<head>
  <meta charset="utf-8">
  <title>Component Lab</title>
  <base href="/">
</head>
<body>
  <component-lab>Loading...</component-lab>
</body>
</html>
```

**Comment by manuelhuber** [comment]
I tried this approach but I get the following error, when I try to run the app with `ng serve --app 1`

`ERROR in Error: ComponentLabModule is not an NgModule
    at _getNgModuleMetadata (C:\Users\ManuelH\_Work\MDE\webportal\node_modules\@angular\compiler-cli\src\ngtools_impl.js:140:15)
    at _extractLazyRoutesFromStaticModule (C:\Users\ManuelH\_Work\MDE\webportal\node_modules\@angular\compiler-cli\src\ngtools_impl.js:109:26)
    at Object.listLazyRoutesOfModule (C:\Users\ManuelH\_Work\MDE\webportal\node_modules\@angular\compiler-cli\src\ngtools_impl.js:53:22)
    at Function.NgTools_InternalApi_NG_2.listLazyRoutes (C:\Users\ManuelH\_Work\MDE\webportal\node_modules\@angular\compiler-cli\src\ngtools_api.js:91:39)
    at AotPlugin._getLazyRoutesFromNgtools (C:\Users\ManuelH\_Work\MDE\webportal\node_modules\@ngtools\webpack\src\plugin.js:207:44)
    at _donePromise.Promise.resolve.then.then.then.then.then (C:\Users\ManuelH\_Work\MDE\webportal\node_modules\@ngtools\webpack\src\plugin.js:443:24)
    at process._tickCallback (internal/process/next_tick.js:109:7)`

I'm using angular cli version 1.4.3 and Typescript 2.5.3

**Comment by brandonroberts** [comment]
Fixed via #24
